static/js/narrow.js

```diff
@@ -88,11 +88,17 @@ exports.activate = function (raw_operators, opts) {
             exports.narrow_title = filter.operands("stream")[0];
         }
     } else if (filter.has_operator("is")) {
-        exports.narrow_title = filter.operands("is")[0];
-    } else if (filter.has_operator("pm-with")) {
-        exports.narrow_title = "private";
-    } else if (filter.has_operator("group-pm-with")) {
-        exports.narrow_title = "private group";
+        var title = filter.operands("is")[0];
+        title = title.charAt(0).toUpperCase() + title.slice(1) + " messages";
+        exports.narrow_title = title;
+    } else if (filter.has_operator("pm-with") || filter.has_operator("group-pm-with")) {
+        var emails = filter.public_operators()[0].operand;
+        var names = people.get_recipients(people.emails_strings_to_user_ids_string(emails));
+        if (filter.has_operator("pm-with")) {
+            exports.narrow_title = names;
+        } else {
+            exports.narrow_title = names + " and others";
```

Comment:
What happens if there's any invalid email addresses here (since it's a search operator, there's no validation) -- does that make it break or something else?